### PR TITLE
Addition to the Stats

### LINKED ISSUE #4619: REST API usage stats
Your Rocket.Chat version: 0.42.0

Hey guys,

i've been wondering if theres an endpoint available to get the usage of rocket chat. E.g. getting the count of registered users.

Kind regards,

### PR DESCRIPTION
**Pull request by graywolf336**
@RocketChat/core 

Done:
- Message Count per Room Type
- Accessing the stats history via the rest api

Closes #4619

### CODE REVIEW

File: server/startup/migrations/v082.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+RocketChat.Migrations.add({
+	version: 82,
+	up: function() {
+		RocketChat.models.Rooms.find().forEach((room) => {
+			RocketChat.models.Rooms.incMsgCountById(room._id, RocketChat.models.Messages.find({ rid: room._id, t: { $exists: true }}).count());
```
Comment (marceloschmidt) [inline server/startup/migrations/v082.js]: Use update and not increment for correctly setting the number of messages.

Comment (graywolf336) [inline server/startup/migrations/v082.js]: Actually after reviewing this, our review together is incorrect as this increments the amount of messages by how many system messages there are.

**Comment by engelgabriel** [comment]
@marceloschmidt is this PR ok now?
@rodrigok can you please review this too?

**Comment by kassanmoor** [comment]
This stats is available in the api? how I can collect data externally?

**Pull request merged**